src/main/kotlin/org/jmailen/gradle/kotlinter/support/RuleSets.kt

```diff
@@ -1,29 +1,29 @@
 package org.jmailen.gradle.kotlinter.support
 
-import com.pinterest.ktlint.core.RuleSet
-import com.pinterest.ktlint.core.RuleSetProvider
+import com.pinterest.ktlint.core.RuleProvider
+import com.pinterest.ktlint.core.RuleSetProviderV2
 import com.pinterest.ktlint.ruleset.experimental.ExperimentalRuleSetProvider
 import java.util.ServiceLoader
-import kotlin.comparisons.compareBy
 
-fun resolveRuleSets(
-    providers: Iterable<RuleSetProvider>,
+internal fun resolveRuleProviders(
+    providers: Iterable<RuleSetProviderV2>,
     includeExperimentalRules: Boolean = false,
-): List<RuleSet> {
-    return providers
-        .filter { includeExperimentalRules || it !is ExperimentalRuleSetProvider }
-        .map { it.get() }
-        .sortedWith(
-            compareBy {
-                when (it.id) {
-                    "standard" -> 0
-                    else -> 1
-                }
-            },
-        )
-}
+): Set<RuleProvider> = providers
```

Comment:
Previously, we've been returning list of sets of rule providers, with the new apis we can return flat collection of rule providers. ktlint's release notes explains the changes in a more elaborate way - [link](https://github.com/pinterest/ktlint/releases/tag/0.47.0)